src/JJMasterData.WebApi/Services/MasterApiService.cs

```diff
@@ -99,73 +97,60 @@ public MasterApiListResponse GetListFields(string elementName, int pag, int regp
 
         return listRet;
     }
-
-    public List<ResponseLetter> SetFields(Hashtable[] listParam, string elementName, bool replace = false)
+    public IEnumerable<ResponseLetter> SetFields(IEnumerable<Hashtable> paramsList, string elementName, bool replace = false)
     {
-        if (listParam == null)
-            throw new ArgumentNullException(nameof(listParam));
+        if (paramsList == null)
+            throw new ArgumentNullException(nameof(paramsList));
 
         var dictionary = GetDataDictionary(elementName);
         if (!dictionary.Api.EnableAdd | !dictionary.Api.EnableUpdate)
             throw new UnauthorizedAccessException();
 
         var formService = GetFormService(dictionary);
-        var listRet = new List<ResponseLetter>();
-        foreach (Hashtable values in listParam)
+
+        foreach (var values in paramsList)
         {
-            ResponseLetter ret = replace ?
+            yield return replace ?
```

Comment:
I prefer the other way

Reply:
Relax with Rider...

Reply:
Increbily this is not a Rider suggestion, I was studying about the `yield` keyword, makes a lot of sense to me.
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/yield

This is what made me a big fan of `yield`
![image](https://user-images.githubusercontent.com/52143624/206067658-7ff56d06-2d4d-4204-b54c-e403af092ceb.png)

When using `yield`, we no longer need the boilerplate `ret` variable we use on foreach.

Reply:
Example of `foreach` without the boilerplate `ret` variable

```cs
var numbers = GetEvenNumbers(5);
foreach (int i in numbers)
{
    Console.WriteLine($"{i}");
}

IEnumerable<int> ProduceEvenNumbers(int upto)
{
    for (int i = 0; i <= upto; i += 2)
    {
        yield return i;
    }
}
```

In Portguese `yield return` means something like "Produza para o meu return"

Reply:
This put a bug in my mind! But I understand now. tks for a example